test/e2e/shared/services.go

```diff
@@ -0,0 +1,183 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package shared
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elb"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachinerytypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateLBService(e2eCtx *E2EContext, svcNamespace string, svcName string, k8sclient crclient.Client) string {
+	Byf("Creating service of type Load Balancer with name: %s under namespace: %s", svcName, svcNamespace)
+	svcSpec := corev1.ServiceSpec{
+		Type: corev1.ServiceTypeLoadBalancer,
+		Ports: []corev1.ServicePort{
+			{
+				Port:     80,
+				Protocol: corev1.ProtocolTCP,
+			},
+		},
+		Selector: map[string]string{
+			"app": "nginx",
+		},
+	}
+	CreateService(svcName, svcNamespace, nil, svcSpec, k8sclient)
+	elbName := ""
+	Eventually(func() bool {
+		svcCreated := &corev1.Service{}
+		err := k8sclient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: svcNamespace, Name: svcName}, svcCreated)
+		Expect(err).NotTo(HaveOccurred())
+		if lbs := len(svcCreated.Status.LoadBalancer.Ingress); lbs > 0 {
+			ingressHostname := svcCreated.Status.LoadBalancer.Ingress[0].Hostname
+			elbName = strings.Split(ingressHostname, "-")[0]
+			return true
+		}
+		return false
+	}, e2eCtx.E2EConfig.GetIntervals("", "wait-create-service")...).Should(BeTrue())
```

Comment:
eks e2e test failed because `wait-create-service` is not defined. I think unmanaged e2e test will also fail.

There are two options, whichever you prefer as I don't have a strong opinion about it.
* Use a specific interval like 60*time.Second
* Define the variable in e2e conf files. As the function is used for both e2e tests, it needs to be defined in [e2e_eks_conf.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/e2e_eks_conf.yaml#L224-L238) and [e2e_eks_conf.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/e2e_conf.yaml#L286-L302)

Reply:
Ah perfect... I was wondering where those intervals were coming from. Thanks so much for solving that mystery for me. 

I'll go with the specific duration option. Just a second.